azurerm/data_source_nat_gateway.go

```diff
@@ -0,0 +1,127 @@
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"idle_timeout_in_minutes": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"public_ip_address_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"public_ip_prefix_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"resource_guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"subnet_ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"zones": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceArmNatGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.NatGatewaysClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Nat Gateway %q (Resource Group %q) was not found", name, resourceGroup)
+		}
+		return fmt.Errorf("Error reading Nat Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", resp.Name)
+	d.Set("sku_name", resp.Sku.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if natGatewayPropertiesFormat := resp.NatGatewayPropertiesFormat; natGatewayPropertiesFormat != nil {
+		d.Set("idle_timeout_in_minutes", natGatewayPropertiesFormat.IdleTimeoutInMinutes)
+		if err := d.Set("public_ip_address_ids", flattenArmNatGatewaySubResourceID(natGatewayPropertiesFormat.PublicIPAddresses)); err != nil {
+			return fmt.Errorf("Error setting `public_ip_address_ids`: %+v", err)
+		}
+		if err := d.Set("public_ip_prefix_ids", flattenArmNatGatewaySubResourceID(natGatewayPropertiesFormat.PublicIPPrefixes)); err != nil {
+			return fmt.Errorf("Error setting `public_ip_prefix_ids`: %+v", err)
+		}
+		d.Set("resource_guid", natGatewayPropertiesFormat.ResourceGUID)
+		if err := d.Set("subnet_ids", flattenArmNatGatewaySubResourceID(natGatewayPropertiesFormat.Subnets)); err != nil {
+			return fmt.Errorf("Error setting `subnet_ids`: %+v", err)
+		}
+	}
+	d.Set("type", resp.Type)
```

Comment:
Is `type` really necessary? Can we remove this?

Reply:
Removed